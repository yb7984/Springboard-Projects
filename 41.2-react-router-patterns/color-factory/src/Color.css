.Color{
    width: 100vw;
    height: 100vh;
    padding-top: 100px;
}