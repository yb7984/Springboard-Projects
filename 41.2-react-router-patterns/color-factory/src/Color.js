import { Link, Redirect, useParams } from 'react-router-dom';
import './Color.css';

const Color = ({ colors = [] }) => {
    const { colorName } = useParams();
    const color = colors.find(color => (color.name === colorName));

    if (color) {
        return (<div className="Color" style={{ backgroundColor: color.value }}>
                <h1>This is {color.name}</h1>
                <h1>Is it beautiful?</h1>
                <h1>
                    <Link to="/colors">Go back!</Link>
                </h1>
            </div >
        );
    } else {
        return <Redirect to="/colors" />;
    }
}

export default Color;