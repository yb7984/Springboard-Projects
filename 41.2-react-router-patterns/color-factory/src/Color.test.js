import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import Color from './Color';

const colors = [
    {
        name: "Red",
        value: "#FF0000"
    }
];

const params = {
    colorName: "Red"
};

//mock the useParams
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: () => params
}));

it('renders without crashing', () => {
    render(<MemoryRouter><Color colors={colors} /></MemoryRouter>);
});


it("matches snapshot, works with the colorName in colors", function () {
    params.colorName = "Red";
    
    const { asFragment, queryByText } = render(
        <MemoryRouter>
            <Color colors={colors} />
        </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();

    expect(queryByText('This is Red')).toBeInTheDocument();
});


it("matches snapshot, works with the colorName not in colors", function () {
    params.colorName = "Blue";
    
    const { asFragment, queryByText } = render(
        <MemoryRouter>
            <Color colors={colors} />
        </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();

    expect(queryByText('This is Red')).not.toBeInTheDocument();
});