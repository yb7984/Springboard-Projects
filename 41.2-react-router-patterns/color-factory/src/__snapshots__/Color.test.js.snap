// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, works with the colorName in colors 1`] = `
<DocumentFragment>
  <div
    class="Color"
    style="background-color: rgb(255, 0, 0);"
  >
    <h1>
      This is Red
    </h1>
    <h1>
      Is it beautiful?
    </h1>
    <h1>
      <a
        href="/colors"
      >
        Go back!
      </a>
    </h1>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot, works with the colorName not in colors 1`] = `<DocumentFragment />`;
